services/cli/manifest-resolver.js

```diff
@@ -85,8 +85,15 @@ class ManifestResolver {
             plugin.optional = false;
           }
 
+          var ok;
+          try {
+            ok = compareVersions(needed[artifactId].version, plugin.version) == 1;
+          } catch (x) {
+            logger.warn(`Cannot compare ${artifactId}:${needed[artifactId].version} to ${plugin.version}: ${x}`);
```

Comment:
Whatever Node library you are using to compare versions, it is hopeless in the face of the version numbers actually used in Jenkins components. We want to be using `jenkinsci/lib-version-number` instead. (As, for example, `incrementals-tools` does.)

Reply:
I BTW, https://issues.jenkins-ci.org/browse/JENKINS-53767 is probably worth looking at. I agree at some point we'd need a central tool to do all this.